scraping_via_selenium_and_soup.py

```diff
@@ -0,0 +1,127 @@
+from selenium.webdriver.chrome.service import Service
+import pandas as pd
+
+base_url = 'https://www.pakwheels.com'
+page_url = '/used-cars/search/-/ct_lahore/ca_college-road/'
+
+
+# load website using chrome webdriver
+def get_driver(website):
+    service = Service('../chromedriver')  # chromedriver path
+    driver = webdriver.Chrome(service=service)
+    driver.get(website)
+
+    return driver
+
+
+def get_soup(driver_for_drive):
+    # get html from driver and make it soup
+    html_data = driver_for_drive.page_source
+    soup = BeautifulSoup(html_data, features="html.parser")
+
+    return soup
+
+
+def get_product_details(delicious_soup):
+    # get product details
+    car_name_data = delicious_soup.find('div', id='scroll_car_info').get_text(separator='\n', strip=True)
+    car_name = car_name_data.split('\n')[0]
+
+    price = delicious_soup.find('strong', class_='generic-green').get_text(separator=' ', strip=True)
+
+    city = delicious_soup.find('p', class_='detail-sub-heading').get_text(separator=' ', strip=True).split()[2]
+
+    location_data = (delicious_soup.find('p', class_='detail-sub-heading')
+                     .get_text(separator=',', strip=True).split(',', maxsplit=2))
+    location = ','.join(location_data[:2])
+
+    model_year = \
+        delicious_soup.find('table', class_='table table-bordered text-center table-engine-detail fs16').get_text(
+            separator=',', strip=True).split(',')[0]
+
+    mileage = delicious_soup.find('table', class_='table table-bordered text-center table-engine-detail fs16').get_text(
+        separator='-', strip=True).split('-')[1]
+
+    seller_name_index = 0
+    seller_name_data = delicious_soup.find('div', class_='owner-detail-main').get_text(separator=',', strip=True).split(
+        ',')
+
+    # checking "is seller 'Dealer' or not? "
+    if seller_name_data[seller_name_index] == 'Dealer:':
+        seller_name_index = 1
+    seller_name = seller_name_data[seller_name_index]
+
+    is_featured = False
+    featured_or_not = delicious_soup.find('div', id='myCarousel').get_text(separator=',', strip=True).split(',')[0]
+    if featured_or_not == 'FEATURED':
+        is_featured = True
+
+    images = delicious_soup.find('ul', class_='lSPager lSGallery').find_all('img')
+    cover_image_url = images[0]['data-original']
+
+    image_1_url = images[1]['data-original']
+    image_2_url = images[2]['data-original'] if len(images) > 2 else None
+
+    return {
+        'car_name': car_name,
+        'price': price,
+        'city': city,
+        'location': location,
+        'model_year': model_year,
+        'mileage': mileage,
+        'seller_name': seller_name,
+        'is_featured': is_featured,
+        'cover_image_url': cover_image_url,
+        'image_1': image_1_url,
+        'image_2': image_2_url
+    }
+
+
+def get_page_urls():
+    page_no = 1
+    page_urls_list = []
+
+    while True:
+        page_urls_list.append(f'{base_url}{page_url}?page={page_no}')
```

Comment:
use more intelligent rules to get the last page and then create list of all the pages here